### DISCUSSION TITLE
Why `podman` command doesn't require a daemon?

### DISCUSSION THREAD

**Discussion by abitrolly** [containers/podman#13058] [Q&A]
Why all external commands need to run `podman system service` to interact with `podman` daemon  while `podman` itself doesn't need it?

**Comment by baude** [comment]
I am not sure what you mean by external commands, but if you want to interact with Podman at an API level (specifically REST), then something has to "serve" that.  The podman cli interacts directly with our libpod APIs in golang.

**Reply by abitrolly** [reply]
By external commands I mean this specific Go lib https://github.com/anchore/stereoscope/pull/87

**Reply by rhatdan** [reply]
Basically external tools can either execute Podman command directly, which is difficult to manage, or they can use the API to a socket activated Podman which is more stable interface.

**Reply by abitrolly** [reply]
That still doesn't answer the question. By executing `podman` directly you mean executing `podman` binary? But the question is now `podman` binary operates on containers without bringing up daemon?

**Reply by rhatdan** [reply]
When my book comes out you will understand.

Podman just pulls an image creates storage, configures the OCI Runtime Specification and launches the OCI runtime (crun, runc, kata ...)
When you run a daemon like Docker or Podman as a socket activation, they
pull an image creates storage, configures the OCI Runtime Specification and launches the OCI runtime (crun, runc, kata ...)

There is nothing special about a daemon. Podman takes advantage of User Namespace to do all of this stuff in rootless mode.

The big difference between Docker is that the Docker Daemon controls multiple clients from interacting with container storage.  While Podman uses the OS (file system locking) to allow multiple Podmans use the storage at the same time.
